kernel/tsc/src/lib.rs

```diff
@@ -7,15 +7,19 @@ use core::sync::atomic::{AtomicUsize, Ordering};
 
 
 #[derive(Debug)]
-pub struct TscTicks(u64);
+pub struct TscTicks(u128);
 
 impl TscTicks {
     /// Converts ticks to nanoseconds. 
     /// Returns None if the TSC tick frequency is unavailable.
-    pub fn to_ns(&self) -> Option<u64> {
-         get_tsc_frequency().ok().map(|freq| {
-            (self.0 * 1000000000) / freq 
-         })
+    pub fn to_ns(&self) -> Option<u128> {
+        if let Some(freq) = get_tsc_frequency().ok() {
+            return self.0.checked_mul(1000000000).map(|checked_tsc| {
+                checked_tsc / freq
+            });
+        }
```

Comment:
Kindly update the docs as well to reflect the new conditions under which `None` may be returned. Also, let's stick with the combinator format, something like this:
```
get_tsc_frequency()
    .ok()
    .and_then(|freq| self.0.checked_mul(1000000000)
        .map(|checked_tsc| checked_tsc / freq)
    )
```